### PR TITLE
Edit step-89.cc.

### PR DESCRIPTION
**Pull request by bangerth**
@jh66637 @peterrum @fdrmrc @kronbichler FYI.

This patch primarily just edits the in-code documentation of step-89 to make it more readable. Over the years, we've found that splitting the code with too many `// ...` comments provides fine-grained documentation (as one would have if you were just reading code you're not familiar with) but makes it difficult to get the "big picture" of what a code does. As a consequence, we comment tutorials differently than we do the rest of the code. This patch merges many such comments together into bigger blocks. There are also other, small text issues it addresses (including making sure everything uses complete sentences).

### CODE REVIEW

**Comment by bangerth** [comment]
Ping?

**Comment by jh66637** [comment]
I am fine with this :)

File: examples/step-89/step-89.cc [hunk 1]
```diff
@@ -1433,21 +1476,26 @@ namespace Step89
...
-        // Materials do not change during the simulation, therefore
-        // there is no need to precompute the values after
-        // the first @c gather_evaluate() again.
         c_r->gather_evaluate(c, EvaluationFlags::values);
         rho_r->gather_evaluate(rho, EvaluationFlags::values);
       }
 
 
-    // Set up inverse mass operator.
+    // Next, we set up inverse mass operator and the acoustic
```
Comment (kronbichler) [inline examples/step-89/step-89.cc:1484]:
```suggestion
    // Next, we set up the inverse mass operator and the acoustic
```

File: examples/step-89/step-89.cc [hunk 2]
```diff
@@ -1433,21 +1476,26 @@ namespace Step89
...
+    // operator. Using `remote_value_type=VectorizedArray<Number>`
+    // makes the operator use point-to-point interpolation. These two
+    // objects are then used to create a `RungeKutta2` object to
+    // perform the time integration.
+    //
+    // We also compute the the maximum speed of sound, needed for the
```
Comment (kronbichler) [inline examples/step-89/step-89.cc:1490]:
```suggestion
    // We also compute the maximum speed of sound, needed for the
```

File: examples/step-89/step-89.cc [hunk 3]
```diff
@@ -1480,14 +1522,13 @@ namespace Step89
...
   }
 
-  // @sect3{Set up and run Nitsche-type mortaring}
-  //
-  // The main purpose of this function is to fill a
-  // `FERemoteEvaluationCommunicator` object that is needed for Nitsche-type
-  // mortaring. Additionally, the corresponding remote evaluators are set up
-  // using this remote communicator. Eventually, the operators are handed to the
-  // time integrator that runs the simulation.
+  // @sect4{Set up and running of the Nitsche-type mortaring scheme}
```
Comment (kronbichler) [inline examples/step-89/step-89.cc:1525]:
```suggestion
  // @sect4{Setup and running of the Nitsche-type mortaring scheme}
```

File: examples/step-89/step-89.cc [hunk 4]
```diff
@@ -1480,14 +1522,13 @@ namespace Step89
...
   //
+  // The alternative to the previous function is to use the mortaring
+  // scheme -- implemented in the following function.  This function
+  // can only be run when deal.II is configured using CGAL (but the
+  // previous function can still be used without CGAL), and so errors
+  // out of CGAL is not available.
```
Comment (kronbichler) [inline examples/step-89/step-89.cc:1531]:
```suggestion
  // out if CGAL is not available.
```

**Review by kronbichler — APPROVED** [review]
This is a very nice update. I spotted a few typos, see below.

**Comment by bangerth** [comment]
Thank you, @kronbichler . I've merged your suggestions!

**Review by kronbichler — APPROVED** [review]

**Pull request merged**